Blog/gst-build-cross-build/cross-building-with-gstreamer.md

```diff
@@ -0,0 +1,320 @@
+
+```
+# sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu libc6-dev-arm64-cross
+```
+
+This is installing a minimal tool-chain in `/usr/aarch64-linux-gnu/`
+
+#### Cross file generated with generate-cross-file.py
+
+Here is the cross file used to build for `aarch64`, this file has been generated with this [helper script](https://gitlab.freedesktop.org/dabrain34/gst-build/blob/dab_add_cross_file_generation/generate-cross-file.py) allowing to generated the cross file for other target as well.
+As you can see, here, we don't use a dedicated rootfs because gst-build will build all that we need for the GStreamer essentials.
+
+```
+# ./generate-cross-file.py
+```
+
+```
+[host_machine]
+system = 'linux'
+cpu_family = 'aarch64'
+cpu = 'arm64'
+endian = 'little'
+
+[properties]
+c_args = ['-Wall', '-g', '-O2']
+cpp_args = ['-Wall', '-g', '-O2']
+objc_args = ['-Wall', '-g', '-O2']
+objcpp_args = []
+c_link_args = []
+cpp_link_args = []
+objc_link_args = []
+objcpp_link_args = []
+pkg_config_libdir = ['__unknown_sysroot__']
+
+
+[binaries]
+c = ['aarch64-linux-gnu-gcc']
+cpp = ['aarch64-linux-gnu-g++']
+ar = ['aarch64-linux-gnu-ar']
+pkgconfig = 'pkg-config'
+cmake = ['false']
+strip = ['aarch64-linux-gnu-strip']
+
+```
+
+Here `meson` will use `aarch64-linux-gnu-gxx` to compile with the given arguments setup above. As `meson` does not recommend to use environment variables, the cross file contains hard-coded path to the sysroot to provide package config.
+Indeed since `meson` > 0.54, you can define `pkg_config_libdir` which will help pkg-config to search for the package configuration files for the given target. You can also tell the path to the pkg-config wrapper by modifying the pkgconfig variable as well.
+Predefined cross file can also be found in `gst-build/data/cross-files`
+
+
+#### Configuring the project for Zynq UltraScale+ MPSoC rev F
+
+When the cross file is ready, we can now configure `gst-build` in order to have a dedicated build for our platform. Here I'm disabling some unnecessary options of `gst-build` such as *libav*, *vaapi* or *gtk_doc*.
+
+Please ensure that you have the last meson version, otherwise gst-build will take glib from the system.
+
+```
+# /path/to/meson_0_54 build-cross-arm64 --cross-file=my-meson-cross-file.txt -D omx=enabled -D sharp=disabled -D gst-omx:header_path=/opt/allegro-vcu-omx-il/omx_header -D gst-omx:target=zynqultrascaleplus -D libav=disabled -D rtsp_server=disabled -D vaapi=disabled -D disable_gst_omx=false -Dugly=disabled -Dgtk_doc=disabled
+
+```
+
+After this step, you should be able to build with `ninja`.
+
+```
+# ninja -C build-cross-arm64
+```
+
+#### Installing
+
+Last but not the least, you need to install the artifacts in a given folder to be mounted by your target with NFS by example. You have to provide a **DESTDIR** variable to `ninja` and it will install in `$DESTDIR/usr/local/` as install prefix.
```

Comment:
What's the difference between using `DESTDIR` and passing a custom prefix property to meson when configuring?

Reply:
DESTDIR is more flexible, that's what I read from the doc.